swarm.go

```diff
@@ -0,0 +1,180 @@
+package httpapi
+
+import (
+	"context"
+	"time"
+
+	"github.com/ipfs/interface-go-ipfs-core"
+	inet "github.com/libp2p/go-libp2p-net"
+	"github.com/libp2p/go-libp2p-peer"
+	"github.com/libp2p/go-libp2p-peerstore"
+	"github.com/libp2p/go-libp2p-protocol"
+	"github.com/multiformats/go-multiaddr"
+)
+
+type SwarmAPI HttpApi
+
+func (api *SwarmAPI) Connect(ctx context.Context, pi peerstore.PeerInfo) error {
+	pidma, err := multiaddr.NewComponent("p2p", pi.ID.Pretty())
+	if err != nil {
+		return err
+	}
+
+	saddrs := make([]string, len(pi.Addrs))
+	for i, addr := range pi.Addrs {
+		saddrs[i] = addr.Encapsulate(pidma).String()
+	}
+
+	return api.core().request("swarm/connect", saddrs...).Exec(ctx, nil)
+}
+
+func (api *SwarmAPI) Disconnect(ctx context.Context, addr multiaddr.Multiaddr) error {
+	return api.core().request("swarm/disconnect", addr.String()).Exec(ctx, nil)
+}
+
+type streamInfo struct {
+	Protocol string
+}
+
+type connInfo struct {
+	Addr       string
+	Peer       string
+	JLatency   time.Duration `json:"Latency"`
+	Muxer      string
+	JDirection inet.Direction `json:"Direction"`
+	JStreams   []streamInfo   `json:"Streams"`
+}
```

Comment:
Another place where I'd rather just parse up-front and introduce an intermediate type.

(again, we can punt on this)